src/redux/profile/profileMiddleware.ts

```diff
@@ -0,0 +1,42 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+
+import { api } from '@/api/api';
+import { onOpenConnection } from '@/redux/connection/connectionSlice';
+import { AppStartListening } from '@/redux/listenerMiddleware';
+
+import { setMenuItemVisible } from '../local/localSlice';
+
+export interface ProfileState {
+  uiPanelVisibility: {
+    // The key here should match the id of the menu item,
+    // else will be ignored
+    [key: string]: boolean;
+  };
+  markNodes: string[];
```

Comment:
Just realized - how about adding the name of the started profile to this topic as well?  

It would be nice to add that information somewhere in the UI - to answer the question of "Oh, I wonder which profile I have running"

Reply:
Making a ticket in notion